src/extensions/default/InlinePaddingEditor/InlinePaddingEditor.js

```diff
@@ -0,0 +1,232 @@
+    var properties          = JSON.parse(require("text!PaddingProperties.json"));
+
+    /** @const @type {number} */
+    var DEFAULT_PADDING  = "5px"; // This is the default value of the padding 
+
+    /** @type {number} Global var used to provide a unique ID for each padding editor instance's _origin field. */
+    var lastOriginId = 1;
+
+    /**
+     * Inline widget containing a paddingEditor control
+     * @param {!string} padding  Initially selected padding
+     * @param {!CodeMirror.TextMarker} marker
+     */
+    function InlinePaddingEditor(padding, marker) {
+        this._padding = padding;
+        this._marker = marker;
+        this._isOwnChange = false;
+        this._isHostChange = false;
+        this._origin = "+InlinePaddingEditor_" + (lastOriginId++);
+
+        this._handlePaddingChange = this._handlePaddingChange.bind(this);
+        this._handleHostDocumentChange = this._handleHostDocumentChange.bind(this);
+
+        InlineWidget.call(this);
+    }
+
+    InlinePaddingEditor.prototype = Object.create(InlineWidget.prototype);
+    InlinePaddingEditor.prototype.constructor = InlinePaddingEditor;
+    InlinePaddingEditor.prototype.parentClass = InlineWidget.prototype;
+
+    /** @type {!paddingEditor} paddingEditor instance */
+    InlinePaddingEditor.prototype.paddingEditor = null;
+
+    /**
+     * Range of code we're attached to; _marker.find() may by null if sync is lost.
+     * @type {!CodeMirror.TextMarker}
+     */
+    InlinePaddingEditor.prototype._marker = null;
+
+    /** @type {boolean} True while we're syncing a paddingEditor change into the code editor */
+    InlinePaddingEditor.prototype._isOwnChange = null;
+
+    /** @type {boolean} True while we're syncing a code editor change into the paddingEditor*/
+    InlinePaddingEditor.prototype._isHostChange = null;
+
+    /** @type {number} ID used to identify edits coming from this inline widget for undo batching */
+    InlinePaddingEditor.prototype._origin = null;
+
+
+    /**
+     * Returns the current text range of the padding value we're attached to, or null if
+     * we've lost sync with what's in the code.
+     * @return {?{start:{line:number, ch:number}, end:{line:number, ch:number}}}
+     */
+    InlinePaddingEditor.prototype.getCurrentRange = function () {
+        var pos, start, end;
+
+        pos = this._marker && this._marker.find();
+
+        start = pos && pos.from;
+        if (!start) {
+            return null;
+        }
+
+        end = pos.to;
+        if (!end) {
+            end = {line: start.line};
+        }
+
+        // Even if we think we have a good range end, we want to run the
+        // regexp match to see if there's a valid match that extends past the marker.
+        // This can happen if the user deletes the end of the existing padding value and then
+        // types some more.
+
+        //Manuelly find the position of the first occurance of padding value in the line
```

Comment:
sp "Manually".  Also, add a space between `//` and the first letter of each line in the comment for readability.